src/views/course_pairings.vue

```diff
@@ -0,0 +1,239 @@
+        specify_term: function() {
+            this.getData();
+        },
+
+        /**
+         * Function is called when column is modified. The data is used
+         * to populate the page with the specified column.
+         */
+        column: function() {
+            this.getData();
+        }
+    },
+    methods: {
+        /**
+         * Function gets course pairing data from API for the specified
+         * course, term and column and calls parseData to parse retuned data.
+         */
+        getData: function() {
+            // Converting to parameter enums
+            var column_name = this.column === "TA" ? "ta" : "instructor";
+
+            var course_where = "";
+            var term_where = "";
+            if (this.specify_term != "All Terms") {
+                term_where = `${this.specify_term}`;
+            }
+            if (this.specify_course != "All Courses") {
+                course_where = `${this.specify_course}`;
+            }
+
+            let url =
+                "{" +
+                '"what":"course_pairings",' +
+                `"user_id":"${this.$route.params.user_id}",` +
+                `"term":"${term_where}",` +
+                `"course_code":"${course_where}",` +
+                `"column":"${column_name}"` +
+                "}";
+
+            url = JSON.parse(url);
+            fetch(generate_get_url(url))
+                .then(res => res.json())
+                .then(data => this.parseData(data))
+                .catch(err => {
+                    this.$emit("error", err.toString());
+                });
+        },
+
+        /**
+         * Function parses data from API call that will be rendered on the
+         * spreadsheet and select component.
+         */
+        parseData: function(data) {
+            var pairings = {};
+            for (let element of data.DATA) {
+                // Create an associative array with keys as course code and values
+                // as an array of instructors/tas for that couse.
+                if (element.course_code in pairings) {
+                    pairings[element.course_code].push({
+                        name: element.name,
+                        user_id: element.user_id
+                    });
+                } else {
+                    pairings[element.course_code] = [
+                        {
+                            name: element.name,
+                            user_id: element.user_id
+                        }
+                    ];
+                }
+
+                // Pushing list of distinct terms
+                if (this.list_terms.indexOf(element.term) < 0) {
+                    this.list_terms.push(element.term);
+                }
+            }
+
+            var user_associations = [];
+            var index_pairings = 0;
+            for (var key in pairings) {
+                // Converting associative arrays to indexed array where first
+                // element is key value
```

Comment:
What are you doing here? I don't understand... `Object.entries(...)` may help.

Reply:
I'm just iterating over the dictionary

Reply:
No, I mean what is the inside of the loop trying to accomplish?

Reply:
It's creating a 2D array. The 1D array contained within is the row displayed on the table. I need to create this 2D array because the original data structure contains a JSON object with the name and user id.

Reply:
I still don't understand. Can you give me an example of an input and an output?

Reply:
The input would be the dictionary:
{["CSC100"]=>[{name:"Richard", user_id: "richard1"}, {name:"Richard", user_id: "richard2"}], 
["CSC101"]=>[{name:"Richard", user_id: "richard1"}, {name:"Richard", user_id: "richard2"}]}

The output would be
[["CSC100", "Richard", "Richard"], ["CSC101", "Richard", "Richard"]]

Reply:
How about

```
flattenUserList = (list) => {
    return list.map((x) => x.name)
}

ret = Object.keys(courseDict).map((course) => flattenUserList(courseDict[course]))
```

does that do what you want?